examples/go-app-counter/main.go

```diff
@@ -0,0 +1,30 @@
+//go:build linux
+// +build linux
+
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+const (
+	DefaultByteCodeFile = "bpf_bpfel.o"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc clang -no-strip -cflags "-O2 -g -Wall" bpf ./bpf/app_counter.c -- -I.:/usr/include/bpf:/usr/include/linux
+func main() {
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+
+	processKprobe(stop)
```

Comment:
Maybe I'm missing something, but the way this seems to work is that we start the Kprobe user space code, but never get to the rest of the userspace functions.  Something would need to signal processKprobe to stop before it would continue to processTracepoint, right?  Could we run each in a separate thread simultaneously?  Or better yet, combine them into one user space app that cycles through the different maps and logs the count for each.

Reply:
I started that way when there were only 2 programs but since u suggested on the other PR cover all or most the userspace would require more work to include all

Reply:
How do you get them all to run, though?

Reply:
I.e., if you did
```
	go processKprobe(stop)
	go processTracepoint(stop)
	go processTC(stop)
	go processUprobe(stop)
	go processXdp(stop)
```
Would they all run in parallel?

Reply:
how best I can test the userspace I think all my validation was done for bc ?